presto-yarn-test/src/test/groovy/com/teradata/presto/yarn/PrestoClusterTest.groovy

```diff
@@ -187,6 +189,31 @@ class PrestoClusterTest
     }
   }
 
+  @Test
+  void 'flex set of workers - multinode-flex-worker'()
+  {
+    PrestoCluster prestoCluster = new PrestoCluster(slider, hdfsClient, 'resources-multinode-single-worker.json', TEMPLATE)
+    prestoCluster.withPrestoCluster {
+      prestoCluster.assertThatPrestoIsUpAndRunning(1)
+      assertThatAllProcessesAreRunning(prestoCluster)
+      def queryExecutor = prestoCluster.queryExecutor
+      waitForNodesToBeActive(queryExecutor, prestoCluster)
+
+      checkState(workers.size() >= 2, "Number of slaves set in the test yaml configuration should be atleast 3")
+      prestoCluster.flex(WORKER_COMPONENT, 2)
+      waitForNodesToBeActive(queryExecutor, prestoCluster)
```

Comment:
actually it might take a while before we can do that check. So then it might fail intermittently if it takes a while to flex the nodes up and running. waitForNodesToBeActive retries until allNodes are up. I can add that prestoCluster.allNodes check after line 204 if you think its ok